connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -1036,7 +1056,8 @@ object SparkSession extends Logging {
    *
    * @since 3.5.0
    */
-  def getDefaultSession: Option[SparkSession] = Option(defaultSession.get())
+  def getDefaultSession: Option[SparkSession] =
+    Option(defaultSession.get()).filterNot(s => s.client != null && s.client.hasSessionChanged)
```

Comment:
I wonder if we can use the same naming with the Python side so we can easily land the same fix at both sides in the future.

Reply:
Sure, I'll look into the Python code, and try to make it much more harmonised with it.

Reply:
There is no Scala counterparts for `is_stopped` or `is_closed` unfortunately, so I named it `isSessionValid` instead (not public).